ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -511,6 +511,21 @@ List<LogSegment> getLogSegments() {
     return result;
   }
 
+  /**
+   * @return count of partially written log segments
+   */
+  int getPartialLogSegmentCount() {
+    int partialLogSegmentCount = 0;
+    LogSegment segment = log.getFirstSegment();
+    while (segment != null) {
+      if (segment.getEndOffset() < segment.getCapacityInBytes() * 0.8) {
+        partialLogSegmentCount++;
+      }
```

Comment:
Can we filter out the active log segment? since this metric is part of compaction. Compaction doesn't care about the last active log segment?
And can we also have a metric to show how many bytes we wasted because of partial log segments?

Reply:
Filtered the journal.
Added the wastedLogSegmentSpace metric.